fe/fe-core/src/main/java/org/apache/doris/tablefunction/MetadataGenerator.java

```diff
@@ -1084,18 +1103,62 @@ private static TFetchSchemaTableDataResult tableOptionsMetadataResult(TSchemaTab
                     }
                     trow.addToColumnValue(new TCell().setIntVal(distributionInfo.getBucketNum())); // BUCKETS_NUM
                     trow.addToColumnValue(new TCell().setIntVal(olapTable.getPartitionNum())); // PARTITION_NUM
+                    dataBatch.add(trow);
+                } // for table
+            } // for db
+        } // for catalog
+        result.setDataBatch(dataBatch);
+        result.setStatus(new TStatus(TStatusCode.OK));
+        return result;
+    }
+
+    private static TFetchSchemaTableDataResult tablePropertiesMetadataResult(TSchemaTableRequestParams params) {
+        if (!params.isSetCurrentUserIdent()) {
+            return errorResult("current user ident is not set.");
+        }
+        TUserIdentity tcurrentUserIdentity = params.getCurrentUserIdent();
+        UserIdentity currentUserIdentity = UserIdentity.fromThrift(tcurrentUserIdentity);
+        TFetchSchemaTableDataResult result = new TFetchSchemaTableDataResult();
+        List<TRow> dataBatch = Lists.newArrayList();
+        List<Long> catalogIds = Env.getCurrentEnv().getCatalogMgr().getCatalogIds();
+        for (Long catalogId : catalogIds) {
```

Comment:
Each catalog has its own `information_schema` database.
So there we should not traverse all catalogs, but only the catalog which this query belongs to

Reply:
> Each catalog has its own `information_schema` database. So there we should not traverse all catalogs, but only the catalog which this query belongs to

thank you. i have handle this case.